stripe/src/main/java/com/stripe/android/googlepay/PaymentsClientFactory.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripe.android.googlepay
+
+import android.app.Activity
+import android.content.Context
+import com.google.android.gms.wallet.PaymentsClient
+import com.google.android.gms.wallet.Wallet
+import com.google.android.gms.wallet.WalletConstants
+
+internal class PaymentsClientFactory(
+    private val context: Context
+) {
+    fun create(
+        environment: StripeGooglePayEnvironment
+    ): PaymentsClient {
+        val options = Wallet.WalletOptions.Builder()
+            .setEnvironment(
+                when (environment) {
+                    StripeGooglePayEnvironment.Production ->
+                        WalletConstants.ENVIRONMENT_PRODUCTION
+                    StripeGooglePayEnvironment.Test ->
+                        WalletConstants.ENVIRONMENT_TEST
+                }
+            )
+            .build()
+
+        return if (context is Activity) {
+            Wallet.getPaymentsClient(context, options)
+        } else {
+            Wallet.getPaymentsClient(context, options)
+        }
```

Comment:
why have the if/else if the body is the same?

Reply:
There are two different methods, one that takes `Activity` and one that takes `Context`. However, I don't think we need to distinguish here and we can always use the `Context` variant.